File: server/src/modules/dashboard/dashboard.controller.ts
import { Context } from 'koa';
import { Result } from '@/core/result';
import dashboardService from './dashboard.service';

class DashboardController {
  /**
   * GET /dashboard/workbench
   * 获取当前登录员工的工作台聚合看板数据
   */
  async workbench(ctx: Context) {
    const userId = (ctx.state as any).user?.id as number;
    if (!userId) {
      ctx.status = 401;
      ctx.body = Result.error('未登录', 401);
      return;
    }
    const data = await dashboardService.workbench(userId);
    ctx.body = Result.success(data);
  }
}

export default new DashboardController();

File: java-server/src/main/java/com/todothink/modules/dashboard/controller/DashboardController.java
package com.todothink.modules.dashboard.controller;

import com.todothink.core.exception.BusinessException;
import com.todothink.core.result.Result;
import com.todothink.core.security.SecurityUtils;
import com.todothink.modules.dashboard.service.DashboardService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/dashboard")
public class DashboardController {

    private final DashboardService dashboardService;

    public DashboardController(DashboardService dashboardService) {
        this.dashboardService = dashboardService;
    }

    @GetMapping("/workbench")
    public Result<?> workbench() {
        Integer userId = SecurityUtils.currentUserId();
        if (userId == null) {
            throw new BusinessException("未登录", 401);
        }
        Map<String, Object> data = dashboardService.workbench(userId);
        return Result.success(data);
    }
}
